src/main/java/com/google/sps/data/User.java

```diff
@@ -0,0 +1,239 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.ArrayList;
+
+/**
+* Represents the static data of a user.
+*/
+public class User {
+    private final String userID;
+    private final String username;
+    private final String firstName;
+    private final String lastName;
+    private final String profilePicture;
+    private final ArrayList<String> gamesCreated;
+    private final int numGamesFinished;
+    private final int numGamesCreated;
+    private final ArrayList<Pair<String, Long>> gamesCompletedWithTime;
+
+    /**
+    * Builder class used to create User objects.
+    */
+    public static class Builder {
+        // Required parameters
+        private final String userID;
+
+        // Optional parameters - initialized to default values
+        private String username = null;
+        private String firstName = null;
+        private String lastName = null;
+        private String profilePicture = null;
+        private ArrayList<String> gamesCreated = null;
```

Comment:
For the List, sometimes it might crash by initializing it to null. For example, it is quite comment that some other function might check the size of the list (i.e. gamesCreated.size()). In this case, setting gamesCreated to null will crash the code. So a better initial value could be an empty list.

private List<String> gamesCreated = new ArrayList<>();

Reply:
I think in this case crashing is a good thing. If we have never explicitly set a value for the list, and someone is trying to access it anyways, then this should be a problem. Crashing is a good way to immediately point out that the list shouldn't be accessed. If it doesn't crash here and instead the client is interacting with an empty list, we have to find out manually why the list is empty.

Setting the list to be initially empty should be something done explicitly outside the class.

Reply:
Actually the code will crash at some random library rather than crash in your code. For example, suppose you use a library to print created game IDs. It will crash at "strings.size()". 

public String printListOfString(List<String> strings) {
   if (strings.size() == 0) {
      return "";
  }
 ... 
}

Also, we should consider the meaning of this class. This class is User.java and some users might not create any games. In this case, the gamesCreated should be initialized to an empty list.

Reply:
Makes sense, I've made similar changes over all the files.